From f830b86ef770100f6e3fa234ac3e00eee7a1cd70 Mon Sep 17 00:00:00 2001
From: Peter Varga <pvarga@inf.u-szeged.hu>
Date: Mon, 16 Nov 2020 11:11:13 +0100
Subject: [PATCH] [Backport] mac: make find_sdk.py work when the sdk goes to 11

Bug: 1098738
Change-Id: I25b84537a445ecb8f80241c98d4753932f5f7c90
Commit-Queue: Nico Weber <thakis@chromium.org>
Commit-Queue: Mark Mentovai <mark@chromium.org>
Auto-Submit: Nico Weber <thakis@chromium.org>
Reviewed-by: Mark Mentovai <mark@chromium.org>
Cr-Commit-Position: refs/heads/master@{#781835}
Reviewed-by: Allan Sandfeld Jensen <allan.jensen@qt.io>
---

diff --git a/chromium/build/mac/find_sdk.py b/chromium/build/mac/find_sdk.py
index 38c2883..a2f7455 100755
--- a/chromium/build/mac/find_sdk.py
+++ b/chromium/build/mac/find_sdk.py
@@ -2,8 +2,7 @@
 # Copyright (c) 2012 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
-
-"""Prints the lowest locally available SDK version greater than or equal to a
+r"""Prints the lowest locally available SDK version greater than or equal to a
 given minimum sdk version to standard output.
 
 If --developer_dir is passed, then the script will use the Xcode toolchain
@@ -14,8 +13,10 @@
 toolchain bin dir.
 
 Usage:
-  python find_sdk.py [--developer_dir DEVELOPER_DIR] [--print_sdk_path] \
-  [--print_bin_path] 10.6  # Ignores SDKs < 10.6
+  python find_sdk.py     \
+      [--print_sdk_path] \
+      [--print_bin_path] \
+      10.6  # Ignores SDKs < 10.6
 
 Sample Output:
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk
@@ -88,7 +89,7 @@
     raise SdkError('Install Xcode, launch it, accept the license ' +
       'agreement, and run `sudo xcode-select -s /path/to/Xcode.app` ' +
       'to continue.')
-  sdks = [re.findall('^MacOSX(10\.\d+)\.sdk$', s) for s in os.listdir(sdk_dir)]
+  sdks = [re.findall('^MacOSX(\d+\.\d+)\.sdk$', s) for s in os.listdir(sdk_dir)]
   sdks = [s[0] for s in sdks if s]  # [['10.5'], ['10.6']] => ['10.5', '10.6']
   sdks = [s for s in sdks  # ['10.5', '10.6'] => ['10.6']
           if parse_version(s) >= parse_version(min_sdk_version)]
